assets/js/components/ClusterDetails/ChecksSelectionNew.test.jsx

```diff
@@ -117,7 +144,7 @@ describe('ClusterDetails ChecksSelectionNew component', () => {
 
     await waitFor(() => screen.getAllByRole('heading'));
 
-    const saveButton = screen.getByRole('button');
+    const saveButton = screen.getByTestId('save-selection-button');
```

Comment:
Did the previous query work the same or you replaced `getByRole` with `getByTestId` because it actually didn't work?

Keep in mind that:

> In the spirit of [the guiding principles](https://testing-library.com/docs/guiding-principles), it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible.

https://testing-library.com/docs/queries/bytestid/